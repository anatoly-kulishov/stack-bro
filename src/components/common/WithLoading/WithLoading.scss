@use 'src/assets/styles/variables' as *;

.loading-overlay {
  top: 0;
  left: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  border-radius: 8px;
  z-index: 200;
}

.spinner-centered {
  position: absolute;
  top: 50%;
  right: 50%;
  transform: translate(50%, -50%);
  z-index: 300;
}

.blurred {
  background-color: rgba(255, 255, 255, .9);
}