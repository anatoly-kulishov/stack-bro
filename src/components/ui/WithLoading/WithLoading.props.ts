export interface IWithLoading {
  isLoading: boolean;
  spinnerSize: string;
}
