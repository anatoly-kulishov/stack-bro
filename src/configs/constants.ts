/**
 * API
 */
export const API_BASE_PATH: string = 'https://social-network.samuraijs.com/api/1.0';
export const API_KEY: string = 'a66e18f8-02d8-41af-affd-0739238765ec';
export const URL_FOR_REGISTRATION: string = 'https://social-network.samuraijs.com/signUp';
export const MOCK_USER_AUTH_DATA = {
  email: 'rewq_gol_47@mail.ru',
  password: 'rewq3511',
};

/**
 * General
 */
export const SITE_TITLE = 'StackBro';
export const SPINNER_SIZE = '40px';
export const DEFAULT_ICONS_SIZE = '24px';
export const MY_LINKEDIN = 'https://www.linkedin.com/in/anatoly-kulishov';
export const MY_GITHUB = 'https://github.com/anatoly-kulishov';
export const smthWentWrong = 'Something went wrong, please reload the page';

/**
 * Buttons
 */
export const LOGIN_BUTTON = 'Log in';
export const LOGIN_AS_GUEST_BUTTON = 'Login as guest';
export const CREATE_NEW_ACCOUNT_BUTTON = 'Create New Account';
export const GO_BACK_BUTTON = 'Go Back';
export const REMEMBER_ME_LABEL = 'Remember me';

/**
 * Validations
 */
export const MIN_PASS_LENGTH: number = 8;
export const MIN_SEARCH_LENGTH: number = 3;
export const FILE_SIZE_LIMIT: number = 10;
export const FILE_SIZE_LIMIT_ERROR: string = 'Size must be less 10Mb';
export const ALLOWED_EXTENSIONS_ERROR: string = 'Not allowed format file!';

/**
 * Placeholders
 */
export const EMAIL_PLACEHOLDER = 'Your email';
export const PASSWORD_PLACEHOLDER = 'Your password';
export const CAPTCHA_PLACEHOLDER = 'Symbols from image';

/**
 * Time
 */
export const TIME_FORMATS = {
  POST_ITEM: 'yyyy-MM-dd HH:mm',
  POST_ITEM_FULL: 'yyyy-MM-dd HH:mm:ss',
};
