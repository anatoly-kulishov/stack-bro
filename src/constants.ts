export const _apiBase = 'http://joblink.quangdel.com/api';
export const __accessToken__ = 'access_token';
export const __refreshToken__ = 'refresh_token';
export const errorMessage = 'Something went wrong';