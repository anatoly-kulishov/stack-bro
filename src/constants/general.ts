export const SITE_TITLE = 'StackBro';