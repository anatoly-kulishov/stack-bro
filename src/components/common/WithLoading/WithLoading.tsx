import React, { FC } from 'react';

import { Spinner } from '../Spinner/Spinner';

import './WithLoading.scss';

export type WithLoadingPropsType = {
  isLoading: boolean;
  spinnerSize: string;
};

export const WithLoading: FC<WithLoadingPropsType> = ({ isLoading, spinnerSize, children }) => {
  return (
    <>
      {isLoading ? (
        <div data-testid="spinner">
          {children}
          <div className="loading-overlay blurred" />
          <div className="spinner-centered">
            <Spinner size={spinnerSize} />
          </div>
        </div>
      ) : (
        children
      )}
    </>
  );
};
