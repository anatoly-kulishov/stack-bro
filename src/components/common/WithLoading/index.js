import WithLoading from "./WithLoading";

export default WithLoading;