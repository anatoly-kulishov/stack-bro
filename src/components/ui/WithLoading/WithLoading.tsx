import React, { FC } from 'react';
import cn from 'classnames';

import { Spinner } from '../Spinner/Spinner';
import { IChildren } from '../../../shared/types';
import { IWithLoading } from './WithLoading.props';
import styles from './WithLoading.module.scss';

export const WithLoading: FC<IWithLoading & IChildren> = ({ isLoading, spinnerSize, children }) => {
  return (
    <>
      {isLoading ? (
        <div data-testid={styles.spinner}>
          {children}
          <div className={cn(styles.loadingOverlay, styles.blurred)} />
          <div className={styles.spinnerCentered}>
            <Spinner size={spinnerSize} />
          </div>
        </div>
      ) : (
        children
      )}
    </>
  );
};
