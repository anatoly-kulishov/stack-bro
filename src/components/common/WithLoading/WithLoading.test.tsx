import React from 'react';
import { render, screen } from '@testing-library/react';

import { WithLoading } from './WithLoading';
import { SPINNER_SIZE } from '../../../constants/general';

describe('WithLoading', () => {
  test('spinner is displayed and dimensions converge', () => {
    render(<WithLoading isLoading={true} spinnerSize={SPINNER_SIZE} />);
    expect(screen.getByTestId('spinner')).toBeInTheDocument();
    expect(screen.getByTestId('spinner')).toMatchSnapshot();
  });

  test("spinner isn't displayed", () => {
    render(<WithLoading isLoading={false} spinnerSize={SPINNER_SIZE} />);
    expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();
  });
});
