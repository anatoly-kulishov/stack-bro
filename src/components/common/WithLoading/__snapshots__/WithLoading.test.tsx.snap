// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTests 1 1`] = `
<div
  data-testid="spinner"
>
  <div
    class="loading-overlay blurred"
  />
  <div
    class="spinner-centered"
  >
    <div
      class="custom-spinner-wrapper"
    >
      <svg
        class="custom-spinner"
        height="40px"
        viewBox="0 0 66 66"
        width="40px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="custom-spinner-path"
          cx="33"
          cy="33"
          fill="none"
          r="30"
          stroke-linecap="round"
          stroke-width="6"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`WithLoading spinner is displayed and dimensions converge 1`] = `
<div
  data-testid="spinner"
>
  <div
    class="loading-overlay blurred"
  />
  <div
    class="spinner-centered"
  >
    <div
      class="custom-spinner-wrapper"
    >
      <svg
        class="custom-spinner"
        height="40px"
        viewBox="0 0 66 66"
        width="40px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="custom-spinner-path"
          cx="33"
          cy="33"
          fill="none"
          r="30"
          stroke-linecap="round"
          stroke-width="6"
        />
      </svg>
    </div>
  </div>
</div>
`;
